x/merkledb/db_test.go

```diff
@@ -40,19 +39,11 @@ func newDB(ctx context.Context, db database.Database, config Config) (*merkleDB,
 }
 
 func newDefaultConfig() Config {
-	return Config{
-		BranchFactor:                BranchFactor16,
-		Hasher:                      DefaultHasher,
-		RootGenConcurrency:          0,
-		HistoryLength:               defaultHistoryLength,
-		ValueNodeCacheSize:          units.MiB,
-		IntermediateNodeCacheSize:   units.MiB,
-		IntermediateWriteBufferSize: units.KiB,
-		IntermediateWriteBatchSize:  256 * units.KiB,
-		Reg:                         prometheus.NewRegistry(),
-		TraceLevel:                  InfoTrace,
-		Tracer:                      trace.Noop,
-	}
+	config := NewConfig()
+	config.HistoryLength = defaultHistoryLength
```

Comment:
Why is this overridden?

Reply:
This was because some of the tests do checks against the history. Maybe a better idea than what we're doing now would be for us to just use `NewConfig` in the tests and manually override there instead of doing it here.

Reply:
Removed this function entirely and tests that rely on the history length override the default config.